docs/src/references/hsm-support.md

```diff
@@ -1,242 +1,228 @@
-
-# Test connection with
-sudo -E -u tedge p11tool --provider p11-kit-client.so --list-tokens
+```sh title="tedge reconnect c8y"
+Disconnecting from Cumulocity
+Removing bridge config file... ✓
+Disabling tedge-mapper-c8y... ✓
+reconnect to Cumulocity cloud.:
+        device id: marcel-hsm-device-rsa
+        cloud profile: <none>
+        cloud host: thin-edge-io.eu-latest.cumulocity.com:8883
+        auth type: Certificate
+        certificate file: /etc/tedge/device-certs/rsa/tedge-certificate.pem
+        cryptoki: true
+        bridge: built-in
+        service manager: systemd
+        mosquitto version: 2.0.20
+        proxy: Not configured
+Creating device in Cumulocity cloud... ✓
+Restarting mosquitto... ✓
+Waiting for mosquitto to be listening for connections... ✓
+Enabling tedge-mapper-c8y... ✓
+Verifying device is connected to cloud... ✓
+Checking Cumulocity is connected to intended tenant... ✓
+Enabling tedge-agent... ✓
 ```
 
-### using it in %%te%%
+`cryptoki: true` in the connection summary confirms that we connected using our PKCS #11 token.
 
-Once we're sure our cryptographic token is accessible, we can use it in %%te%% to connect to
-Cumulocity. In following steps we assume that `c8y.url` is already set.
+## Key selection
 
-13. Build %%te%% with cryptoki support
+<!-- at the moment this isn't tested very extensively -->
 
-The HSM support in %%te%% is still experimental, so it's gated behind `cryptoki` build feature
-switch. You'll have to build %%te%% with that feature and install the built version on your system.
+`tedge` or `tedge-p11-server` will try to find a private key even if the URI is not provided. In
+cases where there are multiple tokens/keys to choose from, the first one returned by the system will
+be automatically selected, but appropriate warning will be emitted:
 
-```sh
-cargo build --features cryptoki
 ```
-
-14. Set required config options
-
-```sh
-# Enable authentication using cryptoki
-tedge config set device.cryptoki.enable true
-# Set the path to the module that will be loaded for authentication
-tedge config set device.cryptoki.module_path /usr/lib/x86_64-linux-gnu/pkcs11/p11-kit-client.so
-# Set the PIN value (123456 is the default)
-tedge config set device.cryptoki.pin 123456
-# Use the built-in bridge (using cryptoki is unsupported with mosquitto)
-tedge config set mqtt.bridge.built_in true
+WARN tedge_p11_server::pkcs11: Multiple keys were found. If the wrong one was chosen, please use a URI that uniquely identifies a key.
 ```
 
-TODO: this doesn't fully work because P11_KIT_SERVER_ADDRESS env var needs to be set for the bridge.
-We don't currently have a config setting for that, so one would have to add this to systemd service
-definition, but as P11 kit usage is subject to change, I'll leave it for later.
-
-15. Reconnect
-
-```sh
-tedge reconnect c8y
-```
-
-## References
-
----
+In such cases, it may be necessary to specify a key.
```

Comment:
How can I do that?

Reply:
added the name of the setting in 8fc4a62b3e